hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/s3guard/DynamoDBMetadataStoreTableManager.java

```diff
@@ -322,6 +326,24 @@ private void createTable(ProvisionedThroughput capacity) throws IOException {
     putVersionMarkerItemToTable();
   }
 
+  /**
+   * Get DynamoDB table server side encryption (SSE) settings from configuration.
+   */
+  private SSESpecification getSseSpecFromConfig() {
+    final SSESpecification sseSpecification = new SSESpecification();
+    sseSpecification.setEnabled(conf.getBoolean(S3GUARD_DDB_TABLE_SSE_ENABLED, false));
+    String cmk = conf.get(S3GUARD_DDB_TABLE_SSE_CMK);
```

Comment:
1. we may want to use getPassword here, probably via s3autils
2. this is just on table creation, isn't it?

Reply:
Yes we should use `S3AUtils::lookupPassword` for this sensitive information.

And yes this is for table creation only, other times in S3Guard, we don't need this config or option. AWS is saying:
> You can switch between the AWS owned CMK, AWS managed CMK, and customer managed CMK at any given time.

So I think in S3Guard, we can actually let user switch from command line. I prefer leaving that as future work, and user can change SSE settings in the AWS way. I assume changing SSE is not a common use case: 1) like tagging which we don't support updating, 2) unlike RCU/WCU which we have a dedicated `SetCapacity` command in S3GuardTool.

Reply:
changing sse could be left to the AWS console. S3Guard CLI can't switch from capacity billing to on-demand, for example (no API call)

Reply:
what's the decision about this change?

Reply:
The decision (between Steve and me) is: in S3Guard command tool we don't support changing encryption algorithm or CMK. Instead, users should go to AWS console (or AWS CLI) for that. S3Guard command tool provisions DynamoDB table if needed, while it is not going to be the best place for DynamoDB table management like billing mode, encryption, tagging etc.

Reply:
(but for getting the CMK, I'll use `S3AUtils::lookupPassword` method)